ocsptools/ocsp_tools.go

```diff
@@ -58,13 +75,34 @@ func GetCertFromIssuerURL(issuerURL string) (*x509.Certificate, error) {
 	return parsedCert, nil
 }
 
-// createOCSPReq creates an OCSP request using either GET or POST (see IETF RFC 6960)
+// GetIssuerCertFromLeafCert takes in a leaf certificate, reads its issuing certificate url field
+// and then calls GetCertFromIssuerURL to return the issuer certificate
+func GetIssuerCertFromLeafCert(leafCert *x509.Certificate) (*x509.Certificate, error) {
+	if len(leafCert.IssuingCertificateURL) == 0 {
+		return nil, fmt.Errorf("Certificate has no issuing certificate url field")
+	}
+
+	issuerURL := leafCert.IssuingCertificateURL[0]
+	issuerCert, err := GetCertFromIssuerURL(issuerURL)
+	if err != nil {
+		return nil, fmt.Errorf("Error getting certificate from issuer url %s: %w", issuerURL, err)
+	}
+
+	return issuerCert, nil
+}
+
+// CreateOCSPReq creates an OCSP request using either GET or POST (see IETF RFC 6960)
 // leafCert is the root certificate (first certificate in the chain)
-// issuerCert is the last certificate in the chain
-// reqMethod is either GET or POST (TODO: change reqMethod to not be string)
+// issuerCert is the certificate of the issuer of the leafCert
+// reqMethod is either GET or POST
+// hash is the hash to use to encode the request (either SHA1 or SHA256 right now)
 func CreateOCSPReq(ocspURL string, leafCert *x509.Certificate, issuerCert *x509.Certificate, reqMethod string, hash crypto.Hash) (*http.Request, error) {
 	if ocspURL == "" {
-		ocspURL = leafCert.OCSPServer[0]
+		// leafCert probably is an intermediary, which may not be required to have an OCSP responder
```

Comment:
This also goes beyond the 80 character line length, can it be broken into a 2-line comment.